### ISSUE TITLE
Routing error for spreeapi

### ISSUE THREAD

**Issue by funkdified** [spree/spree#5198]
I am testing spree locally and get the following error when attempting to choose product options:

    Started GET "/spreeapi/option_types?q%5Bname_cont%5D=&_=1407867241807" for 127.0.0.1 at 2014-08-12 13:14:04 -0500
    
    ActionController::RoutingError (No route matches [GET] "/spreeapi/option_types"):

Here is the relevant part of my `gemfile.lock`

    GIT
      remote: git://github.com/spree/spree.git
      revision: 9d4617334c872e591839a930c2aa52c834ce0896
      branch: 2-3-stable
      specs:
        spree (2.3.2.beta)
          spree_api (= 2.3.2.beta)

And `routes.rb`

      mount Spree::Core::Engine, :at => '/'

**Comment by CamJN** [comment]
As I said, it doesn't work for me in 2.4 (2.4.2 now).

**Comment by JDutil** [comment]
@CamJN it works fine for me on 2-4-stable.  Please provide steps to reproduce the problem or an example application that reproduces the problem along with the relevant information requested in our [Contributing Guidelines](https://github.com/spree/spree/blob/master/CONTRIBUTING.md#filing-an-issue) if people would like to look into it further.

**Comment by CamJN** [comment]
I guess I have to retract my statement. When I created a minimal setup I couldn't reproduce. Must have been something that didn't update properly in my normal project.

**Comment by JDutil** [comment]
I'm closing this issue out since it's fixed on 2.4+, and 2.3 will only be receiving major bug/security fixes now.

**Comment by joshleitzel** [comment]
I've just come across this issue and I think the problem might be in the case where you've initially mounted to `/` (Spree default install) and afterward change it to `/store` or a different path. I inadvertently fixed the issue when I attempted to debug the [`@mountedAt` function](https://github.com/spree/spree/blob/d83733d996241e10906f703a845dfaae08625cbd/core/app/assets/javascripts/spree.js.coffee.erb), which is taken into account when generating the API paths. By editing the file I caused the ERB to re-compile which picked up my new mount path. I think it probably compiled the ERB once, when it was still mounted at `/`, and doesn't re-compile unless *the ERB file changes*. So after changing your mount point you should make a small change to `spree.js.coffee.erb`, which will cause the asset pipeline to re-compile it with the correct mount path.

You can find this file by running:
```
echo "`bundle show spree_core`/app/assets/javascripts/spree.js.coffee.erb"
```
Hope this helps someone in the future who stumbles across this issue!
